src/plugins/template/src/infer_request.cpp

```diff
@@ -0,0 +1,239 @@
+#include "openvino/core/except.hpp"
+#include "openvino/runtime/profiling_info.hpp"
+#include "plugin.hpp"
+#include "template_itt.hpp"
+
+using Time = std::chrono::high_resolution_clock;
+
+namespace {
+
+void allocate_tensor_impl(ov::Tensor& tensor, const ov::element::Type& element_type, const ov::Shape& shape) {
+    if (!tensor || tensor.get_element_type() != element_type) {
+        tensor = ov::Tensor(element_type, shape);
+    } else {
+        tensor.set_shape(shape);
+    }
+}
+
+}  // namespace
+
+// ! [infer_request:ctor]
+TemplatePlugin::InferRequest::InferRequest(const std::shared_ptr<const TemplatePlugin::CompiledModel>& model)
+    : ov::ISyncInferRequest(model) {
+    // TODO: allocate infer request device and host buffers if needed, fill actual list of profiling tasks
+
+    auto requestID = std::to_string(get_template_model()->_requestId.fetch_add(1));
+
+    std::string name = get_template_model()->m_model->get_friendly_name() + "_Req" + requestID;
+    m_profiling_task = {
+        openvino::itt::handle("Template" + std::to_string(get_template_model()->_cfg.deviceId) + "_" + name +
+                              "_Preprocess"),
+        openvino::itt::handle("Template" + std::to_string(get_template_model()->_cfg.deviceId) + "_" + name +
+                              "_Postprocess"),
+        openvino::itt::handle("Template" + std::to_string(get_template_model()->_cfg.deviceId) + "_" + name +
+                              "_StartPipeline"),
+        openvino::itt::handle("Template" + std::to_string(get_template_model()->_cfg.deviceId) + "_" + name +
+                              "_WaitPipline"),
+    };
+
+    m_executable = get_template_model()->get_template_plugin()->_backend->compile(get_template_model()->m_model);
+
+    // Allocate plugin backend specific memory handles
+    m_plugin_input_tensors.resize(get_inputs().size());
+    m_plugin_output_tensors.resize(get_outputs().size());
+
+    // Allocate input/output tensors
+    for (const auto& input : get_inputs()) {
+        allocate_tensor(input, [input](ov::Tensor& tensor) {
+            // Can add a check to avoid double work in case of shared tensors
+            allocate_tensor_impl(tensor,
+                                 input.get_element_type(),
+                                 input.get_partial_shape().is_dynamic() ? ov::Shape{0} : input.get_shape());
+        });
+    }
+    for (const auto& output : get_outputs()) {
+        allocate_tensor(output, [output](ov::Tensor& tensor) {
+            // Can add a check to avoid double work in case of shared tensors
+            allocate_tensor_impl(tensor,
+                                 output.get_element_type(),
+                                 output.get_partial_shape().is_dynamic() ? ov::Shape{0} : output.get_shape());
+        });
+    }
+}
+// ! [infer_request:ctor]
+
+std::vector<ov::VariableState> TemplatePlugin::InferRequest::query_state() const {
+    OPENVINO_NOT_IMPLEMENTED;
```

Comment:
is it technically possible to implement this method or we need to refactor interpreter backend>